asio/include/asio/detail/macos_fenced_block.hpp

```diff
@@ -19,6 +19,7 @@
 
 #if defined(__MACH__) && defined(__APPLE__)
 
+#include <atomic>
 #include <libkern/OSAtomic.h>
```

Comment:
FWIW this header is no longer needed (since you removed all uses of `OSMemoryBarrier()`)